### PR TITLE
[FLINK-30585][flamegraph] Collect the flame graph of a single subtask instead of all subtasks when viewing the flame graph of a single subtask

### PR DESCRIPTION
**Pull request by 1996fanrui**
## What is the purpose of the change

After [FLINK-30185](https://issues.apache.org/jira/browse/FLINK-30185) , we can view the flame graph of subtask level. However, it always collects flame graphs for all subtasks.

We should collect the flame graph of single subtask instead of all subtasks.

## Brief change log

[FLINK-30585][flamegraph] Collect the flame graph of a single subtask instead of all subtasks when viewing the flame graph of a single subtask

### Flame graph collection 

- The stats collected for job vertex and execution vertex will be collected into the executionVertexStatsCache.
- The stats collected for job vertex will be collected into the jobVertexStatsCache.

### Flame graph query

- The stats of job vertex will query from the executionVertexStatsCache.
- The stats of execution vertex will query from the executionVertexStatsCache.

## Verifying this change

This change added tests and can be verified as follows:

  - *Improved all old tests inside of the `VertexThreadInfoTrackerTest`*
  - *Added the `VertexThreadInfoTrackerTest#testExecutionVertexCachedStatsUpdatedAfterRefreshInterval `*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive):no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector:no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented?  not documented

### CODE REVIEW

**Comment by 1996fanrui** [comment]
Hi @reswqa  @pnowojski , would you mind help review this PR in your free time? thanks~

File: flink-runtime/src/test/java/org/apache/flink/runtime/webmonitor/threadinfo/VertexThreadInfoTrackerTest.java [hunk 1]
```diff
@@ -159,106 +188,242 @@ public void testCachedStatsUpdatedAfterRefreshInterval() throws Exception {
...
         // verify that we get the second result on the next request
         Optional<VertexThreadInfoStats> result =
-                tracker.getVertexStats(JOB_ID, EXECUTION_JOB_VERTEX);
+                tracker.getJobVertexStats(JOB_ID, EXECUTION_JOB_VERTEX);
         assertExpectedEqualsReceived(threadInfoStatsAfterRefresh, result);
     }
 
+    /** Tests that cached execution vertex result is NOT reused after refresh interval. */
+    @Test
+    public void testExecutionVertexCachedStatsUpdatedAfterRefreshInterval() throws Exception {
```
Comment (reswqa) [inline flink-runtime/src/test/java/org/apache/flink/runtime/webmonitor/threadinfo/VertexThreadInfoTrackerTest.java:222]: I haven't looked at the previous tests, so I want to know if they can cover the following situations:

When there are pending `JobVertex` granularity request,`ExecutionVertex` granularity request will not be sent.

**Review by reswqa — COMMENTED** [review]
Thanks @1996fanrui for bring up this! This changes almost looks good to me, I only left some minor comments.

**Comment by 1996fanrui** [comment]
Hi @reswqa , thanks a lot for your review.

> Have we manually submitted a job and verified this through `WebUI`?

Yes, I have tested on my Mac through WebUI.

**Comment by 1996fanrui** [comment]
@flinkbot run azure

**Comment by 1996fanrui** [comment]
@flinkbot run azure

**Comment by 1996fanrui** [comment]
The CI fails twice due to the HiveITCase, I have created the FLINK-32294 to follow it.

**Comment by 1996fanrui** [comment]
CI passed, merging~

**Pull request merged**